File: benchmarks/language/file_io/r/write_filtered_fasta.R
library(Biostrings)
records <- readDNAStringSet("data/sequences.fa")
filtered <- records[width(records) >= 2000]
writeXStringSet(filtered, "data/filtered_output.fa")
cat(sprintf("Input records: %d\n", length(records)))
cat(sprintf("Filtered records: %d\n", length(filtered)))
cat(sprintf("Written: %d\n", length(filtered)))

File: benchmarks/language/file_io/python/write_filtered_fasta.py
"""File I/O: Read FASTA, filter by length, write filtered output"""
from Bio import SeqIO

records = list(SeqIO.parse("data/sequences.fa", "fasta"))
filtered = [r for r in records if len(r.seq) >= 2000]
count = SeqIO.write(filtered, "data/filtered_output.fa", "fasta")
print(f"Input records: {len(records)}")
print(f"Filtered records: {len(filtered)}")
print(f"Written: {count}")
